hoot-core/src/main/cpp/hoot/core/io/OsmGeoJsonWriter.cpp

```diff
@@ -0,0 +1,385 @@
+    type = "Point";
+  if (temp_nodes.size() < 4 && type.compare("Polygon") == 0)
+    type = "LineString";
+  if (type.compare("Polygon") == 0 && temp_nodes[0] != temp_nodes[temp_nodes.size() - 1])
+    type = "LineString";
+  _writeKvp("type", type.c_str());
+  _write(",");
+  _write("\"coordinates\": ");
+  bool point = (type.compare("Point") == 0);
+  bool polygon = (type.compare("Polygon") == 0);
+  if (!point)
+    _write("[");
+  if (polygon)
+    _write("[");
+  bool first = true;
+  for (vector<long>::const_iterator it = temp_nodes.begin(); it != temp_nodes.end(); ++it)
+  {
+    if (first)
+      first = false;
+    else
+      _write(",");
+    ConstNodePtr n = _map->getNode(*it);
+    _write(QString("[%1, %2]").arg(QString::number(n->getX(), 'g', _precision)).arg(QString::number(n->getY(), 'g', _precision)));
+  }
+  if (polygon)
+    _write("]");
+  if (!point)
+    _write("]");
+}
+
+void OsmGeoJsonWriter::_writeFeature(ConstElementPtr e)
+{
+  _writeKvp("type", "Feature"); _write(",");
+  _writeKvp("id", QString::number(e->getId())); _write(",");
+  _write("\"properties\": {");
+  _writeKvp("type", _typeName(e->getElementType()));
+  if (_hasTags(e))
+  {
+    _write(",");
+    _writeTags(e);
+  }
+  if (e->getElementType() == ElementType::Relation)
+  {
+    _write(",");
+    _writeRelationInfo(boost::dynamic_pointer_cast<const Relation>(e));
+  }
+  _write("}");
+}
+
+void OsmGeoJsonWriter::_writeNodes()
+{
+  NoInformationCriterion filter;
+  QList<long> nids;
+  const NodeMap& nodes = _map->getNodes();
+  for (NodeMap::const_iterator it = nodes.begin(); it != nodes.end(); ++it)
+  {
+    if (filter.isNotSatisfied(_map->getNode(it->first)))
+      nids.append(it->first);
+  }
+  // sort the values to give consistent results.
+  qSort(nids.begin(), nids.end(), qGreater<long>());
+  for (int i = 0; i < nids.size(); i++)
+  {
+    if (_firstElement)
+      _firstElement = false;
+    else
+      _write(",", true);
+    _writeNode(_map->getNode(nids[i]));
+  }
+}
+
+void OsmGeoJsonWriter::_writeNode(ConstNodePtr node)
+{
+  if (node.get() == NULL)
+    return;
+  _write("{");
+  _writeFeature(node);
+  _write(",");
+  _write("\"geometry\": {");
+  _writeGeometry(node);
+  _write("}");
+  _write("}", false);
+}
+
+void OsmGeoJsonWriter::_writeWays()
+{
+  const WayMap& ways = _map->getWays();
+  for (WayMap::const_iterator it = ways.begin(); it != ways.end(); ++it)
+  {
+    ConstWayPtr w = it->second;
+    //  Skip any ways that have parents
+    set<ElementId> parents = _map->getParents(w->getElementId());
+    if (parents.size() > 0)
+      continue;
+    if (w.get() == NULL)
+      continue;
+    if (_firstElement)
+      _firstElement = false;
+    else
+      _write(",", true);
+    //  Make sure that building ways are "complete"
+    const vector<long>& nodes = w->getNodeIds();
+    bool valid = true;
+    if (OsmSchema::getInstance().isBuilding(w) ||
+        OsmSchema::getInstance().isBuildingPart(w))
```

Comment:
Let me look at it but I think that you are correct.  That makes more sense because we want all areas and not just buildings.